pydantic/datetime_parse.py

```diff
@@ -90,7 +90,10 @@ def parse_date(value: Union[date, StrIntFloat]) -> date:
     Raise ValueError if the input isn't well formatted.
     """
     if isinstance(value, date):
-        return value
+        if isinstance(value, datetime):
```

Comment:
Note that we might want to update the type hint for `value` to accept `datetime` as well.